File: 1/1-7.hs
sqrtIter :: Double -> Double -> Double
sqrtIter guess x
    | abs (guess * guess - x) < 0.001 = guess
    | otherwise = sqrtIter ((guess + x / guess) / 2) x

squareRoot :: Double -> Double
squareRoot x = sqrtIter 1.0 x

main :: IO ()
main = print (squareRoot 2.0)

File: 1/1-7.rs
fn sqrt_iter(guess: f64, x: f64) -> f64 {
    if (guess * guess - x).abs() < 0.001 {
        guess
    } else {
        sqrt_iter((guess + x / guess) / 2.0, x)
    }
}

fn square_root(x: f64) -> f64 {
    sqrt_iter(1.0, x)
}

fn main() {
    println!("{}", square_root(2.0))
}